packages/instrumentation-mongodb/src/instrumentation.ts

```diff
@@ -952,28 +952,26 @@ export class MongoDBInstrumentation extends InstrumentationBase<MongoDBInstrumen
 
   private _defaultDbStatementSerializer(commandObj: Record<string, unknown>) {
     const { enhancedDatabaseReporting } = this.getConfig();
-    const resultObj = enhancedDatabaseReporting
-      ? commandObj
-      : this._scrubStatement(commandObj);
-    return JSON.stringify(resultObj);
-  }
 
-  private _scrubStatement(value: unknown): unknown {
-    if (Array.isArray(value)) {
-      return value.map(element => this._scrubStatement(element));
+    if (enhancedDatabaseReporting) {
+      return JSON.stringify(commandObj);
     }
 
-    if (typeof value === 'object' && value !== null) {
-      return Object.fromEntries(
-        Object.entries(value).map(([key, element]) => [
-          key,
-          this._scrubStatement(element),
-        ])
-      );
-    }
+    const getReplacer = (): Replacer => {
```

Comment:
Micro-optimization: I would hoist it as a private method of current class, so that it is not re-declared on each call to `_defaultDbStatementSerializer()`

Reply:
yeah, let's go